llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -482,11 +481,10 @@ InstructionCost RISCVTTIImpl::getShuffleCost(TTI::ShuffleKind Kind,
       //   vmsne.vi  v0, v8, 0
```

Comment:
Unrelated codegen opportunity

 vmv.x.s a0, v8
 andi    a0, a0, 1
 vmv.v.x v8, a0

Could be a vrgather.vi